### PR TITLE
Addresses issue #283 by adding oauth2 security scopes to operation docs

### PR DESCRIPTION
**Pull request by jimmyjames**
This PR attempts to address issue discussed in issue #283 

Operations often have a specific security scope associated with them, and currently that information is not available via ReDoc. This PR:

- Renders a tooltip to the right of the operation header if there are OAuth2 security definitions for that operation.
- In order to support HTML display in the tooltip, changes dependency on [hint.css](https://github.com/chinchang/hint.css) to [html-hint.css](https://github.com/istarkov/html-hint) (based on hint.css). 

Some things to consider/review from those with more knowledge:

- Other tooltips using hint appear to work OK (probably because html-hint is based on hint), but could use some consideration from domain-experts.
- This pass only supports OAuth2 security definitions per operation as a first-pass - it does not consider api_keys. OK for a first pass?
- This implementation does not attempt to handle any `$ref` to security definitions per-operation. I don't know if this is supported/common so worth considering (perhaps at least needs to be handled for the `securityDefinitions` lookup of the descriptions?).
- Currently displays as `!` styled similar to other tips like Parameters. Might be better to display as some sort of padlock icon (like Swagger UI) to be clearer, but per issue comments starting with the `!`

### CODE REVIEW

File: lib/utils/spec-manager.ts [hunk 1]
```diff
@@ -137,6 +137,42 @@ export class SpecManager {
     return obj;
   }
 
+  getOperationScopes(operationPtr:string) {
+    let that = this;
+    function getSecurityDefinition(name:string, scope:string) {
```
Comment (RomanGotsiy) [inline lib/utils/spec-manager.ts]: Transform this function to fat-arrow function and remove `let that = this` trick:

```js
const getSecurityDefinition = (name: string, scope: string) => {
  // just use `this` here
}
```

**Comment by wilsonge** [comment]
@jimmyjames are you planning on picking up the feedback changes here? I need these for a project so would be happy to pick up if you aren't planning to

**Comment by jimmyjames** [comment]
Hey @wilsonge and @RomanGotsiy sorry for the late reply. @wilsonge if you have some bandwidth and can incorporate the feedback changes that would be great - it may be a little bit until I can get back to this. Thanks!

> I will test these changes on some major documentations powered by ReDoc and most likely will request some more changes.

@RomanGotsiy did you get a chance to do this?

**Comment by wilsonge** [comment]
@jimmyjames See https://github.com/jimmyjames/ReDoc/pull/1

**Comment by jimmyjames** [comment]
I'm not able to pull the changes and test in the next couple days; @wilsonge can you include a screenshot of how it will look? 

Given the desire to not use html-hint, I'm wondering using tooltip is not the right approach at all (unless another solution using hints that supports html content is included), and instead just include the scopes as a subsection under the operation description. It definitely makes the OAuth2 scopes more prominent in the docs instead of tucked away, for better or worse.

@RomanGotsiy curious for your feedback, as well as if you had a chance to test further.

**Comment by RomanGotsiy** [comment]
Hey @jimmyjames,

sorry for delaying this for so long time. I was on vacation.
Now I'm back and I hope we will merge it soon!

Today I will release a new version of ReDoc (as there are a few fixes already merged) and then I will start testing this.
I aim to have this feature merged in the next release!

> Given the desire to not use html-hint, I'm wondering using tooltip is not the right approach at all 
Maybe we'll have to go back to `html-hint`. Will see. Give me a few days and we will merge it finally!

**Comment by RomanGotsiy** [comment]
Hey @jimmyjames,

would you mind if I introduce this feature a bit later together with the rewrite to React (#327)?
This way I will have fewer things to migrate.

**Comment by jimmyjames** [comment]
@RomanGotsiy sorry I missed this until now.

I understand wanting to minimize the amount of migration done. In our case, our API has many different scopes, and our most of our operations require different scopes. Not having the scopes required per operation shown in the docs is not acceptable for us, but we can fork/modify/host ourselves if we need to until migration is done and the feature is added.

**Comment by RomanGotsiy** [comment]
@jimmyjames sorry for not answering for so long...
As far as you may know I'm working on React rewrite of Redoc (https://github.com/Rebilly/ReDoc/pull/357)

So basic version of this feature is already implemented there.
Hope you can help to improve it :)

Closing this PR in favor of React version.
Thanks!

**Pull request closed without merging**